ui_tests/pages/profiles_page.py

```diff
@@ -0,0 +1,299 @@
+        logger.info(f"click delete profile button -> {profile_name}")
+        delete_btn_for_profile_with_name_xpath = f"//button[@data-test='sc4snmp:profile-row-delete' and ancestor::tr//td[text()='{profile_name}']]"
+        delete_btn = driver.find_element(
+            By.XPATH, delete_btn_for_profile_with_name_xpath
+        )
+        delete_btn.click()
+        time.sleep(1)
+
+    def _confirm_delete_profile(self):
+        confirm_delete_xpath = (
+            "//button[@data-test='sc4snmp:delete-modal:delete-button']"
+        )
+        confirm_btn = driver.find_element(By.XPATH, confirm_delete_xpath)
+        confirm_btn.click()
+        time.sleep(1)
+
+    def close_profile_delete_popup(self):
+        logger.info(f"Closing profile delete popup")
+        close_profile_delete_popup_btn_xpath = (
+            "//button[@data-test='sc4snmp:errors-modal:cancel-button']"
+        )
+        close_btn = driver.find_element(By.XPATH, close_profile_delete_popup_btn_xpath)
+        close_btn.click()
+        time.sleep(1)
+
+    def get_profile_type_for_profile_entry(self, profile_name):
+        logger.info(f"getting profile type for profile {profile_name}")
+        profile_type_for_profile_with_name_xpath = f"//td[@data-test='sc4snmp:profile-type' and ancestor::tr//td[text()='{profile_name}']]"
+        profile_type = driver.find_element(
+            By.XPATH, profile_type_for_profile_with_name_xpath
+        )
+        return profile_type.text
+
+    def set_smart_profile_field(self, field_value):
+        logger.info(f"Setting smart profile field {field_value}")
+        smart_profile_field_xpath = (
+            "//div[@data-test='sc4snmp:form:condition-field-input']//span//input"
+        )
+        field = driver.find_element(By.XPATH, smart_profile_field_xpath)
+        field.send_keys(field_value)
+
+    def add_smart_profile_pattern(self, pattern):
+        logger.info(f"Add smart profile pattern {pattern}")
+        add_pattern_button_xpath = "//span[contains(text(),'Add pattern')]"
+        add_pattern_button = driver.find_element(By.XPATH, add_pattern_button_xpath)
+        add_pattern_button.click()
+        time.sleep(1)
+        pattern_row_xpath = (
+            "//div[@data-test='sc4snmp:form:field-pattern']//span//input"
+        )
+        pattern_rows = driver.find_elements(By.XPATH, pattern_row_xpath)
+        pattern_rows[-1].send_keys(pattern)
+
+    def check_if_frequency_setting_field_is_visible(self):
+        logger.info(f"Checking if frequency setting field is visible")
+        xpath = "//div[@data-test='sc4snmp:form:frequency-input']//span//input"
+        try:
+            freq_field = driver.find_element(By.XPATH, xpath)
+            return freq_field.is_displayed()
+        except Exception as e:
+            return False
+
+    def add_condition(self, field_value, operation, value):
+        logger.info(f"Checking if frequency setting field is visible")
```

Comment:
I think this log message is copied from `check_if_frequency_setting_field_is_visible`. Here it should rather be something about adding condition

Reply:
fixed